src/main/kube-auth-proxy/kube-auth-proxy-cert.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { access, mkdir, writeFile } from "fs/promises";
+import path from "path";
+import * as selfsigned from "selfsigned";
+
+export function getKubeAuthProxyCertificate(): selfsigned.SelfSignedCert {
+  const opts = [
+    { name: "commonName", value: "Lens Certificate Authority" },
+    { name: "organizationName", value: "Lens" },
+  ];
+
+  return selfsigned.generate(opts, {
```

Comment:
Not 100% deterministic because cert creation always creates an unique certificate. That said, it has deterministic fields... not sure where we should draw the line here.